### ISSUE TITLE
Label the SDK's issues

### ISSUE THREAD

**Issue by bidoubiwa** [meilisearch/integration-guides#186]
# Issues on label

## Current

As of now we have 
- Labels on issues in the [integration-guide issues](https://github.com/meilisearch/integration-guides/blob/a1db1f728a12e25826a877ec2a9bfa7a3b7e1ecc/resources/issues.md#issue-labeling-in-integration-guides).
- Labels on [SDK's PR's](https://github.com/meilisearch/integration-guides/blob/a1db1f728a12e25826a877ec2a9bfa7a3b7e1ecc/resources/issues.md#issue-labeling-in-integration-guides).

We do not have any rules or process on labelling the issues. As I remember from the past we tried to create issues labels but ultimately we dropped them because they linked to a `project board` which was flooded and we did not think they were useful.

## Why do we want SDK's issues today

There are two main reasons why the subject has been brought up during the weekly of the @meilisearch/integration-team .

- We want to use the `question` label more often to help the @meilisearch/devrel-team find the issues on which they might response.
- We want to know for ourselves at first sight what type of issue we are dealing so we can know, for example, why an issue is stalled (`need-help`, `needs investigation`, ...).

## What are the cons

One big con is that it will require discipline into adding issues. A good example is the labels from the PR's that are still forgotten sometimes.

It will also require in every SDK to add the labels and to maybe update the existing issues with the good label. 

## Solution

There is no real solution to the first con. But forgetting to add a label to an issue is not very problematic as it is not used in any process (yet we should still thrive to make it work)
And yes, we would need to add them manually to every repo unless there is a way to automate that.  

## First iteration

As a first iteration these may be interesting labels to consider adding in our issues process: 
Often used labels have a 🔥 emoji. 

Already exists:
- `question` - Used when a question must be answered by supports.
- `bug` - Used when a bug is raised in an issue 🔥. 
- `good first issue` - Used when the issue can be fixed simply 🔥.
- `documentation` - Used when additional documentation is asked for in the issue. 
- `enhancement` - Used for features and general enhancement 🔥.
- `help wanted` - Used when external help is needed to solve the issue 🔥.
- `spam` - Used before closing to point out the spammy nature. 
- `invalid` - (not sure) used when the issues raises something that is not linked to the repo

First batch:
- `needs investigation` - Used when the issue requires a deep investigation to understand the issue.
- `Tests` - Used when the issue specifically asks to add more tests on a specific part of the code
- `feedback` - Used when the user gives its opinion on the SDK. Not sure if this is used before closing, or really needed.
- `Needs more information` - Used when we need more information to understand the problem

Later batches : 
- `Discussion` - Used when discussing before deciding something ?
- `Can't reproduce` - Used when we can not reproduce the error
- `Needs more information` - Used when we need more information to understand the problem
- `Wont fix` - Used when we do not know how to fix the bug. Maybe redundant with the `bug` labe
- `Refactor` - Used when the issue specifically asks for a part of the code to be refactored (maybe redundant with `enhancement`)

## Questions

I first would like your opinion on
- Is it a good idea to implement this process
- Is it a process that we want to enforce or not. To see if it often used
- Is it something that we first want to use on one specific repo before adding it everywhere

Then: 
- What are the labels we want to use 
- What are the labels we want to add in the first batch

**Comment by curquiza** [comment]
My 2 cents on the core side. The labels I use to regularly clean the issues in the meilisearch repository
- `support` (your kind of `question` equivalent). I use `support` and not `question` because most of the issue starts like "this is broken!" and the final state is "oh I miss used Meilisearch, sorry". This is not a question, not a bug either, but this is support. In `support` scope you have the questions + people coming without any questions, thinking there is a bug, but this is not a bug.
- `needs more info`: when I need more info to classify the issue (99% of the time: is it just `support` or a real `bug`?)

Regarding `support` and `needs more info`, if I don't get an answer in 2 or 3 weeks after having asked precision, I close the issue.

Also, on the core side, we regroup `tests` and `refactor` into `maintenance`, but I cannot affirm this is the right way to do it.
We also have `tooling` label for anything non-related to the search engine but we still provide, like the docker image, homebrew, binaries, the download-latest.sh script...

So the labels I use the most are: `need mode info`, `support`, `bug`, `enhancement`, `good first issue`, `tooling`, `maintenance` and any label related to our specific usage (`tokenizer`, `milli`, `performance`, `windows`). It really helps me a lot managing the repo! So I agree this is a good idea to start using them every time on the integration side. However, it should not replace the answer to the contributor, putting a label is not enough, at least say "thanks for the bug report".

**Comment by curquiza** [comment]
Also, I would add the core team does not manage the product side, so no feature requests are managed in the Meilisearch repo. Which is not the same regarding the integration team. One of the challenges of the integration team is to manage all the feature requests, by filtering them, approving them or not, checking the ones that need more investigation before being implemented...
Of course, you might want to start simple and improve this part of the process later :)

**Comment by ferdi05** [comment]
I don't have much to say, but the DevRel team would be really happy to take care of all the issues labeled with `support` cc @oluademola

**Comment by brunoocasali** [comment]
Thanks for raising this @bidoubiwa!

I really enjoyed all the POV everyone placed here, it was really helpful!
The first problem I would like to solve is to adhere to the process and add labels to the PRs to create a good track of the health of our integrations!

What about: 

- Add the label `needs triage` automatically as @sanders41 suggested.
- Rename the `question` label to `support` in order to start an integration draft between DevRel and Integrations teams. This could work exactly like @curquiza mentioned in her first message.
But we still need to understand better how to handle an issue with this label, should DevRel act before the Integrations or otherwise? Maybe we need a `needs devrel`/`team: devrel` to avoid unintentional rework, two people from different teams working on the same problem without knowing. One more thing, could we have a kind of support that only integrations could do?
- Add `needs more information` because it is common to have incomplete issues.
- Add a `stale` to be used by some automation bot (more explanation below)

So the process to handle the issues could be something like:

- A new issue comes in with the automated label (needs triage).
- The Integrations team (mainly the scope owner) make the first contact with the contributor saying hello and asking for more info if needed, and checking if this could be handled by DevRel.
    - **If this is support:**
        - Add the labels `needs devrel` and `support`, maybe we should add a comment `hey DevRel team, we need your help here`. And then DevRel will handle it 😃
        - If DevRel discovers a bug during the support process, a new comment could be done asking for help directly from the scope owner.
    - **If this is a bug/enhancement or anything not related to support:**
        - Add the labels `needs more info` if needed and another label like `enhancement`, `bug` to describe what kind of problem we are dealing with.
- After the solution, we close the issue (of course 😄).

We could try to automate issue-closing if no further action was done by the author for X days. 
Eg.: The issue has a label `needs more information` (which means the contributor needs to give more info), and the bot adds a label `stale`. 
And if X more days pass without action, the bot closes the issue. The `stale` label could be used just for us to know in advance when the bot closed the issue historical purposes.

What do you guys think?

**Comment by alallema** [comment]
I really like this new process. I just have a few comments:
- I don't think that the first contact has to be the integration team IF it's obvious or sure that the question is for the DevRel team, because the purpose of the support is to save time for the team integration and secondly because I'm afraid that multiple interactions could be confusing and could also flood the issue.
- I really like the `need investigation` flag can we have it?

**Comment by brunoocasali** [comment]
Hi @alallema thanks for your inputs!

> * I don't think that the first contact has to be the integration team IF it's obvious or sure that the question is for the DevRel team because the purpose of the support is to save time for the team integration, and secondly, because I'm afraid that multiple interactions could be confusing and could also flood the issue.

My concern is that way could lead us to a place where we (as integrations) think DevRel is taking care of the issue, and they believe the same about us! In other words, no one is taking care of it in that case.
Or is your idea just regarding the first interaction about the hello stuff?
Because we can add the correct label and just ask DevRel for help through the comments, I prefer to keep the `triage` on our side at this first moment since we already do that.
What do you think @alallema ?

> * I really like the `need investigation` flag can we have it?

Oh yes! But for me the `needs investigation` has the same purpose as `needs triage`, do you think otherwise?

**Comment by alallema** [comment]
> Because we can add the correct label and just ask DevRel for help through the comments, I prefer to keep the triage on our side at this first moment since we already do that.

Of course, I also prefer the triage on your side and my comment meant that we had put the label `DevRel` but also that we had tell them or ping them as they prefer.

**Comment by ferdi05** [comment]
> * But we still need to understand better how to handle an issue with this label, should DevRel act before the Integrations or otherwise? Maybe we need a `needs devrel`/`team: devrel` to avoid unintentional rework, two people from different teams working on the same problem without knowing. One more thing, could we have a kind of support that only integrations could do?

@brunoocasali it's actually something we need to work on. My expectation is that @oluademola will now be able to read most of the support requests from our users, wherever they're coming from. This includes GitHub issues.
I think his role will be to handle such requests before the integration team does, but who will apply the `support` label?

> One more thing, could we have a kind of support that only integrations could do?

Indeed, that's when we need escalating for issues that the DevRel team is unable to solve

**Comment by bidoubiwa** [comment]
This what we (I think I lost my notes) decided for the first itteration: 

1. As much as possible we should use the same tags in all repositories
2. Custom labels may be created in repositories with specific needs, for example "interface-bug" on front-end integrations. 
3. Labels are in try-mode. Meaning that if they are not used it is not a big issue as the objective is to facilitate the maintainers work.
4. The tags that are common to all repo's for the moment are: 
<details>
<summary>The default ones</summary>
- question
- bug
- good first issue
- documentation
- enhancement
- help wanted
</details>

The first batch:
 
- `needs investigation`:  color: `#006B75` text: `Needs to take time to understand the issue.`
- `needs more info` could you give us the HTML color code @curquiza ? text: `This issue needs a minimal complete and verifiable example`
-  `tests`: color: `#50AF71` text: `Request to add some tests.`
- `support`: color: `#CE33FF` text: `Question that may be answered by the support team`
- `stale`: color: `#A2A0D6` text: `Pull request or issue that has recieve no activity for a long time.`

Did I forgot one ? @alallema @brunoocasali 

If we agree on the labels could you thumbs up 👍 this comment ?

> but who will apply the support label?

I think is the responsibility of the integration team

> One more thing, could we have a kind of support that only integrations could do?

We could use `Question` in this case. In the case we put `support` on a complex question, maybe @oluademola can also contact us to tell us if he can't answer so that we do it or if he only needs additional information from us?

**Comment by alallema** [comment]
> Did I forgot one ? @alallema @brunoocasali

I think for the first batch it's great! I like the `stale` one. Maybe `discussion` could be great but not mandatory for now.

**Comment by bidoubiwa** [comment]
@mdubus  What was maintenance for? I'm sorry I forgot :/

**Comment by mdubus** [comment]
@bidoubiwa it was for stuff like eslint, prettier, storybook, package.json ... Mainly for settings, but I think we could also put the tests inside this label

**Comment by curquiza** [comment]
> @bidoubiwa it was for stuff like eslint, prettier, storybook, package.json ... Mainly for settings, but I think we could also put the tests inside this label

In the core team we use `maintenance` for the tests/benchmarks, formating... but also for the refactoring.

**Comment by bidoubiwa** [comment]
`maintenance` is better than `test` I agree! 

New proposal: 
- `needs investigation`:  color: `#006B75` text: `Needs to take time to understand the issue.`
- `needs more info` color: `#5319e7` text: `This issue needs a minimal complete and verifiable example`
-  `maintenance`: color: `#DBA7A9` text: `Issue about maintenance (CI, tests, refacto...)`
- `support`: color: `#F78FB2` text: `Issues related to support questions`
- `stale`: color: `#A2A0D6` text: `Pull request or issue that has recieve no activity for a long time.`

If you agree : 🚀

**Comment by brunoocasali** [comment]
I created issues in all the integration repositories 😸. Let's start using these labels, and we'll see how it goes! 

Thanks for every input here. You're the best people! ❤️

**Comment by bidoubiwa** [comment]
Should we write it down somewhere in the integration guides ?

**Comment by curquiza** [comment]
I agree with @bidoubiwa, could be added to this guide: https://github.com/meilisearch/integration-guides/blob/main/resources/issues.md, WDYT?

**Comment by bidoubiwa** [comment]
I'm re-opening it until we made the guide and the issues
